File: object_relations/app/models/customer.rb
class Customer

  attr_accessor :name
  @@all = []

  def initialize(name)
    @name = name
    @@all << self
  end

  def add_review(text, restaurant)
    review = Review.new(text, restaurant, self)
  end

  def self.all
    @@all
  end

  def self.find_by_name(name)
    self.all.find {|customer| customer.name == name}
  end
end

# customer = customer.add_review('it was good', Restaurant.new)
# a new review tied ot the restaurant
# and tied to the customer

File: javascript/models/customer.js
class Customer {
  constructor(name){
    this.name = name
    store.customers.push(this)
  }

  add_review(text, restaurant){
    var cust = this
    var rev = new Review(text, restaurant, cust)
  }

  restaurants() {
    var revs = this.reviews()
    var rests = []
    revs.forEach((review) => {
      if (rests.includes(review.restaurant)){
          return
        } else {
          rests.push(review.restaurant)
        }
      }
    )
    return rests
  }

  reviews(){
    var filter_reviews = (review) => {
      return review.customer === this
    }
    var revs = store.reviews.filter(filter_reviews)
    return revs
  }

}
